packages/react-native-web/src/exports/Modal/ModalAnimation.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) Nicolas Gallagher.
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import { useEffect, useCallback } from 'react';
+
+import StyleSheet from '../StyleSheet';
+import createElement from '../createElement';
+
+const ANIMATION_DURATION = 300;
+
+function getAnimationStyle(animationType, visible) {
+  if (animationType === 'slide') {
+    return visible ? animatedSlideInStyles : animatedSlideOutStyles;
+  }
+
+  if (animationType === 'fade') {
+    return visible ? animatedFadeInStyles : animatedFadeOutStyles;
+  }
+
+  return visible ? styles.container : styles.hidden;
+}
+
+export type ModalAnimationProps = {|
+  children?: any,
+  animationType?: ?('none' | 'slide' | 'fade'),
+  visible?: ?boolean,
+  onShow?: ?() => void,
+  onDismiss?: ?() => void
+|};
+
+function ModalAnimation(props: ModalAnimationProps) {
+  const {
+    children,
+    animationType,
+    visible,
+    onShow,
+    onDismiss
+  } = props;
+
+  const isAnimated = animationType && animationType !== 'none';
+
+  const animationEndCallback = useCallback(() => {
+    if (visible) {
+      // If animation completed and we're visible,
+      // fire off the onShow callback
+      if (onShow) {
+        onShow();
+      }
+    } else {
+      // If animation completed and we're visible,
+      // fire off the onDismiss callback
+      if (onDismiss) {
+        onDismiss();
+      }
+    }
+  }, [onDismiss, onShow, visible]);
+
+  // If the `visible` flag is changing we want to set the rendering flag to true
+  // before the animations ever will start
+  useEffect(() =>  {
+    if (!isAnimated) {
+      // If !isAnimated we have to manually call `animationEndCallback` - if we
+      // don't it will never get called
+      animationEndCallback();
+    }
+  }, [isAnimated, visible, animationEndCallback]);
```

Comment:
`visible` is in the deps by not used by this effect

Reply:
Right - this is to force it to be called any time the `visible` flag changes.   Perhaps a better comment or a `ref` that it can compare against to prevent it from calling too many times?

Reply:
It's probably a sign that this and the surrounding code should be refactored to maintain correct dependencies

Reply:
I've added a ref to keep track of the previous visibility to compare it to the current & also ended up adding back another check for the `visible` flag to prevent the flash of fade / slide on an initial render of modals that aren't visible.

Let me know if that works for you.